### ISSUE TITLE
New Build failure on AArch64

### ISSUE THREAD

**Issue by AmyWignall-arm** [oneapi-src/oneDNN#1662]
# Summary
This change: https://github.com/oneapi-src/oneDNN/commit/a99bd65d0d24de509d6405d0e798360c26f47794 Has introduced a build failure for us (for acl and non-acl builds):
```
In file included from /home/amywig01/oneDNN/src/cpu/aarch64/jit_sve_512_1x1_convolution.cpp:25:
/home/amywig01/oneDNN/src/cpu/aarch64/jit_sve_512_1x1_convolution.hpp: In instantiation of ‘struct dnnl::impl::cpu::aarch64::jit_sve_512_1x1_convolution_fwd_t<dnnl_f32>::pd_t’:
/home/amywig01/oneDNN/src/cpu/aarch64/jit_sve_512_1x1_convolution.cpp:441:17:   required from here
/home/amywig01/oneDNN/src/cpu/aarch64/jit_sve_512_1x1_convolution.hpp:88:30: error: ‘const memory_desc_t* dnnl::impl::cpu::aarch64::jit_sve_512_1x1_convolution_fwd_t<src_type, wei_type, dst_type>::pd_t::dst_md(int) const [with dnnl_data_type_t src_type = dnnl_f32; dnnl_data_type_t wei_type = dnnl_f32; dnnl_data_type_t dst_type = dnnl_f32; dnnl::impl::memory_desc_t = dnnl_memory_desc]’ marked ‘override’, but does not override
   88 |         const memory_desc_t *dst_md(int index = 0) const override {
      |                              ^~~~~~
make[2]: *** [src/cpu/aarch64/CMakeFiles/dnnl_cpu_aarch64.dir/build.make:391: src/cpu/aarch64/CMakeFiles/dnnl_cpu_aarch64.dir/jit_sve_512_1x1_convolution.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /home/amywig01/oneDNN/src/cpu/cpu_convolution_list.cpp:59:
/home/amywig01/oneDNN/src/cpu/aarch64/jit_sve_512_1x1_convolution.hpp: In instantiation of ‘struct dnnl::impl::cpu::aarch64::jit_sve_512_1x1_convolution_fwd_t<dnnl_f32>::pd_t’:
/usr/include/c++/9/type_traits:1294:12:   required from ‘struct std::is_base_of<dnnl::impl::primitive_desc_t, dnnl::impl::cpu::aarch64::jit_sve_512_1x1_convolution_fwd_t<dnnl_f32>::pd_t>’
/home/amywig01/oneDNN/src/common/impl_list_item.hpp:71:68:   required from ‘constexpr dnnl::impl::impl_list_item_t::type_deduction_helper_t<pd_t>::type_deduction_helper_t() [with pd_t = dnnl::impl::cpu::aarch64::jit_sve_512_1x1_convolution_fwd_t<dnnl_f32>::pd_t]’
/home/amywig01/oneDNN/src/cpu/cpu_convolution_list.cpp:81:86:   required from here
/home/amywig01/oneDNN/src/cpu/aarch64/jit_sve_512_1x1_convolution.hpp:88:30: error: ‘const memory_desc_t* dnnl::impl::cpu::aarch64::jit_sve_512_1x1_convolution_fwd_t<src_type, wei_type, dst_type>::pd_t::dst_md(int) const [with dnnl_data_type_t src_type = dnnl_f32; dnnl_data_type_t wei_type = dnnl_f32; dnnl_data_type_t dst_type = dnnl_f32; dnnl::impl::memory_desc_t = dnnl_memory_desc]’ marked ‘override’, but does not override
   88 |         const memory_desc_t *dst_md(int index = 0) const override {
      |                              ^~~~~~
make[2]: *** [src/cpu/CMakeFiles/dnnl_cpu.dir/build.make:144: src/cpu/CMakeFiles/dnnl_cpu.dir/cpu_convolution_list.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:657: src/cpu/aarch64/CMakeFiles/dnnl_cpu_aarch64.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:630: src/cpu/CMakeFiles/dnnl_cpu.dir/all] Error 2
make: *** [Makefile:144: all] Error 2
```

I noticed that other files in this commit have the change:
```
-        const memory_desc_t *dst_md(int index = 0) const override {
-            return jcp_.with_dw_conv ? dw_conv_pd_->dst_md(index) : &dst_md_;
+        const memory_desc_t *dst_md(
+               int index = 0, bool user_input = false) const override {
+           return jcp_.with_dw_conv
+                   ? dw_conv_pd_->dst_md(index, user_input)
+                   : cpu_convolution_fwd_pd_t::dst_md(index, user_input);
+       }
```
Whereas the file causing the error, `jit_sve_512_1x1_convolution.hpp` does not. If I add this change to that file the the build completes without issue. Maybe this is the bug?

# Version
`onednn_verbose,info,oneDNN v3.2.0 (commit a99bd65d0d24de509d6405d0e798360c26f47794)`

# Environment
```
$ lscpu
Architecture:                    aarch64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
CPU(s):                          64
On-line CPU(s) list:             0-63
Thread(s) per core:              1
Core(s) per socket:              64
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       ARM
Model:                           1
Model name:                      Neoverse-N1
Stepping:                        r3p1
BogoMIPS:                        243.75
L1d cache:                       4 MiB
L1i cache:                       4 MiB
L2 cache:                        64 MiB
L3 cache:                        32 MiB
NUMA node0 CPU(s):               0-63
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Not affected
Vulnerability Retbleed:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:        Mitigation; __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; CSV2, BHB
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp ssbs
```
* OS version (`uname -a`)
```
$ uname -a
Linux ip-10-252-30-169 5.15.0-1036-aws #40~20.04.1-Ubuntu SMP Mon Apr 24 00:20:54 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux
```
* Compiler version (`gcc --version`)
```
$ gcc --version
gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
* CMake version (`cmake --version`)
```
$ cmake --version
cmake version 3.16.3

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```
* CMake output log
See error above

* git hash (`git log -1 --format=%H`)
```
$ git log -1 --format=%H
a99bd65d0d24de509d6405d0e798360c26f47794

```

# Steps to reproduce
Build with, for example:
```
cmake .. -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DCMAKE_BUILD_TYPE=Debug -DDNNL_BUILD_FOR_CI=ON -DONEDNN_BUILD_GRAPH=0 -DDNNL_ENABLE_JIT_PROFILING=0
make -j
```

# Observed behavior
Build fails with error shown above

# Expected behavior
Build completes without error

**Comment by AmyWignall-arm** [comment]
@kawakami-k

**Comment by kawakami-k** [comment]
The following modifications may be necessary.

```
diff --git a/src/cpu/aarch64/jit_sve_512_1x1_convolution.hpp b/src/cpu/aarch64/jit_sve_512_1x1_convolution.hpp
index f3acdb94c..ac7b1f2b1 100644
--- a/src/cpu/aarch64/jit_sve_512_1x1_convolution.hpp
+++ b/src/cpu/aarch64/jit_sve_512_1x1_convolution.hpp
@@ -85,8 +85,11 @@ struct jit_sve_512_1x1_convolution_fwd_t : public primitive_t {
             return status::success;
         }

 

-        const memory_desc_t *dst_md(int index = 0) const override {
-            return jcp_.with_dw_conv ? dw_conv_pd_->dst_md(index) : &dst_md_;
+        const memory_desc_t *dst_md(
+                int index = 0, bool user_input = false) const override {
+            return jcp_.with_dw_conv
+                    ? dw_conv_pd_->dst_md(index, user_input)
+                    : cpu_convolution_fwd_pd_t::dst_md(index, user_input);
         }

 

         const memory_desc_t *arg_md(
```

**Comment by dzarukin** [comment]
Yes, that's exactly the fix coming to the master and rls-v3.2 branches.
